src/firebase-messaging-sw.ts

```diff
@@ -24,43 +20,32 @@ export const firebaseApp = initializeApp({
 // messages. Need to keep this alive
 const messaging = getMessaging(firebaseApp)
 
-const getDbSymkeyStore = async () => {
-  const db = await openDB('w3i-sw-db', 3, {
-    upgrade(database) {
-      const exists = database.objectStoreNames.contains(SYMKEY_OBJ_STORE)
-      if (!exists) {
-        database.createObjectStore(SYMKEY_OBJ_STORE)
-      }
-    }
-  })
-  return db
-}
-
 const getSymKey = async (topic: string) => {
-  const db = await getDbSymkeyStore()
+  const [getSymKeyUsingTopic] = await getDbSymkeyStore()
 
-  const result: string = await db.get(SYMKEY_OBJ_STORE, topic)
+  const result: string = await getSymKeyUsingTopic(topic)
 
   if (result) {
     return result
   }
 
-  throw new Error('No symkey exists for such topic')
+  throw new Error(`No symkey exists for such topic: ${topic}`)
 }
 
-self.addEventListener('push', async ev => {
-  const { blob: encoded, topic } = ev.data!.json().data
+onBackgroundMessage(messaging, async ev => {
+  const encoded = ev.data?.blob
+  const topic = ev.data?.topic
+
+  if (!encoded || !topic) {
```

Comment:
Error case or does this happen in expected ways? Would be good to log when this happens if it shouldn't.

Reply:
This has not happened, it's jut that data is typed as typed with a `?`. I guess I could clean this up to `if (!data)` but this seemed cleaner to me since semantically `data` does not guarantee `blob` and `topic` (at the type level)

Reply:
Added a console.error that can be viewed via devtools. I believe this is sufficient but I can look into adding sentry/mixpanel for this later on (not sure its worth blocking this PR), just not sure how well they'd play with the service worker environment